comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go

```diff
@@ -255,9 +258,38 @@ func (ia *inventoryagent) initData() {
 
 	ia.data["feature_dynamic_instrumentation_enabled"] = getBoolSysProbe("dynamic_instrumentation.enabled")
 
+	// ECS Fargate
+	ia.fetchECSFargateAgentMetadata()
+
 	ia.refreshMetadata()
 }
 
+// fetchECSFargateAgentMetadata fetches ECS Fargate agent metadata from the ECS metadata V2 service.
+// Times out after 5 seconds to avoid blocking the agent startup.
+func (ia *inventoryagent) fetchECSFargateAgentMetadata() {
+	ctx, cc := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cc()
+
+	if !env.IsECSFargate() {
+		return
+	}
+	client, err := ecsmeta.V2()
+	if err != nil {
+		ia.log.Warnf("error while initializing ECS metadata V2 client: %s", err)
+		return
+	}
+
+	// Use the task ARN as hostname
+	taskMeta, err := client.GetTask(ctx)
+	if err != nil {
+		ia.log.Warnf("error while fetching ECS Fargate metadata V2 task: %s", err)
+		return
+	}
+
+	ia.data["ecs_fargate_task_arn"] = taskMeta.TaskARN
+	ia.data["ecs_fargate_cluster_name"] = taskMeta.ClusterName
```

Comment:
Could you add these informations in the [README](https://github.com/DataDog/datadog-agent/blob/main/comp/metadata/inventoryagent/README.md) please 🙏

Reply:
Good catch, I forgot about these entries. Added them in [669e13c](https://github.com/DataDog/datadog-agent/pull/22542/commits/669e13cc9d969857f3cebc16944d252a7dffb1a0)